### ISSUE TITLE
Issues when page system name contains special characters

### ISSUE THREAD

**Issue by jahia-ci** [Jahia/jcontent#1492]
> Imported from Jira, on Thu Jan 30, 2025
> Issue: [BACKLOG-21971](https://jira.jahia.org/browse/BACKLOG-21971) in project: DX BACKLOG
> Priority: Major Type: Bug
> Reporter: @sdusolle (Sonia Dusolle)
> Assignee: _None found in Jira, making reporter the assignee in GitHub_
> Created: Fri Dec 1, 2023, last updated: Mon Jan 13, 2025
> Status: Open
>
> **Links**:
> - relates to: [BACKLOG-21846 - Blank page when accessing a page with special characters in page builder](https://github.com/Jahia/jira-archives/issues/6496) *[[JIRA](https://jira.jahia.org/browse/BACKLOG-21846)]*
> - relates to: [QA-14040 - Unable to create link to content with system name that contains special characters.](https://github.com/Jahia/jira-archives/issues/41594) *[[JIRA](https://jira.jahia.org/browse/QA-14040)]*
>
>
> **Attachments**:
> - [image-2023-12-01-16-23-40-974.png](https://jira-artifacts.jahia.com/jira/BACKLOG/BACKLOG-21971/image-2023-12-01-16-23-40-974.png) _uploaded by: @sdusolle (Sonia Dusolle) (83.1 KB - Dec 1, 2023_)
> - [image-2023-12-01-16-24-32-145.png](https://jira-artifacts.jahia.com/jira/BACKLOG/BACKLOG-21971/image-2023-12-01-16-24-32-145.png) _uploaded by: @sdusolle (Sonia Dusolle) (28.19 KB - Dec 1, 2023_)
>

### <ins>**1st issue**</ins>
* create a rich text with a link to a page with system name = @#$<sup>&€§¢ª¶ø°£™¥‰œæÀ-ž--

--> we cannot save in CE, we get an error saying "The link /sites/digitall/home/newsroom/@ is broken"

[image-2023-12-01-16-24-32-145\.png](https://jira-artifacts.jahia.com/jira/BACKLOG/BACKLOG-21971/image-2023-12-01-16-24-32-145.png)
### <ins>**2nd issue**</ins>
* try to access a page in preview or live mode with system name = @#$</sup>&€§¢ª¶ø°£™¥‰œæÀ-ž-- 

--> 404 error

[image-2023-12-01-16-23-40-974\.png](https://jira-artifacts.jahia.com/jira/BACKLOG/BACKLOG-21971/image-2023-12-01-16-23-40-974.png)
### Current Behavior

We cannot access a page with system name containing special characters in live or preview.

We cannot create a link to this page inside a rich text neither.
### Desired Behavior

Access the page in live and preview mode.

Create a link to it inside a rich text.
## NOTE

Access in edit mode to this kind of page has been fixed with recent tickets : BACKLOG-21846 and QA-14040

**Comment by gflores-jahia** [comment]
Wait for @TheKlem to assess if we want to tackle in the sprint. Leaving this out for now.

**Comment by AKarmanov** [comment]
Hi, the issue here is actually one specific char `#` it is used in navigation and routing by many frontend frameworks, it should normally come at the very end of your working url (after .html). In this case it's in he middle of the url. This behaviour was introduced here: https://github.com/Jahia/jahia-private/pull/1294. 

https://github.com/user-attachments/assets/2babbe8e-9f1a-4d9f-9f02-e89493c3fe7b

Furthermore, for the url in preview, the `#` part starts a fragment, so the request going in is terminated at `@`:

<img width="1589" height="691" alt="Image" src="https://github.com/user-attachments/assets/b5675e3d-06f7-48f7-841a-137aba62d5bd" />

in other words, Jahia doesn't even see the full URL due to browser behaviour.

So the conclusion here is that it is behaving as expected and it just requires user discretion to make sure that urls are constructed correctly when it comes to the usage of `?` and `#`.

**Comment by sdusolle** [comment]
I am fine to say it is the expected behavior - especially as it only happens if we "force" the systemname :
`This issue appears only if we "force" the systemname to include the special characters. If I set the title page to "@#$^&€§¢ª¶ø°£™¥‰œæÀ-ž--" then the systemname is replaced by default by "a-o-TM-oeaea-z".`
